horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/SchemaServiceImpl.java

```diff
@@ -683,16 +688,17 @@ public Integer addOrUpdateLabel(int schemaId, Label labelDTO) {
       LabelDAO label = LabelMapper.to(labelDTO);
       if (label.id == null || label.id < 0) {
          label.id = null;
-         label.schema = em.getReference(SchemaDAO.class, schemaId);
+
+         label.schema = (SchemaDAO) SchemaDAO.findByIdOptional(schemaId)
+               .orElseThrow(() -> ServiceException.notFound("Schema " + schemaId + " not found"));
+
          checkSameName(label);
          label.persistAndFlush();
          emitLabelChanged(label.id, schemaId);
       } else {
-         LabelDAO existing = LabelDAO.findById(label.id);
-         if (existing == null) {
-            label.id = -1;
-            existing = label;
-         }
+         LabelDAO existing = (LabelDAO) LabelDAO.findByIdOptional(label.id)
+               .orElseThrow(() -> ServiceException.notFound("Label " + label.id + " not found"));
+
```

Comment:
This changes behaviour slightly, before if "existing" was not found, it was set to label, and label.idset to -1

Reply:
That's true, but actually that case was not working at all. 

It was setting `existing = label` and then Hibernate would have thrown an exception as the entity it was trying to persist was not managed/associated to the Hibernate session (I think because the id should not be set to `-1` but to `null`).

Given that it was not working before, IMO to be coherent wrt other services if the provided id is not null and there is no entity with that id we should simply fail. Otherwise if we have user wanting to update a label, set the wrong id, it would end up creating a new label which is not the intended behavior. wdyt?